src/pages/api/addItem.ts

```diff
@@ -0,0 +1,98 @@
+import { NextApiRequest, NextApiResponse } from "next";
+import { z, ZodError } from "zod";
+import { Items, List, Prisma, PrismaClient } from "@prisma/client";
+
+const prisma = new PrismaClient();
+
+/* The point of this endpoint is to add an item to a list (someone has typed in and pressed enter).
+First check the name given against the database.
+Since there are multiple ways to spell something we are trying to remove 
+a couple of letters at a time and see if there is a match
+When shorter than 3 letters we just put it in the Other list
+*/
+export default async function handler(
+  request: NextApiRequest,
+  response: NextApiResponse
+) {
+  if (request.method === "POST") {
+    try {
+      const { query, inputList } = inputQueryTest.parse(request.body);
+      let products: Items[] = [];
+      // Getting ID of list we will add product to.  If none there create one
+      const listToBe = await prisma.list.upsert({
+        create: {
+          name: inputList,
+        },
+        update: {},
```

Comment:
why is update empty? Looks like you don't need an upsert here, more like an insert()

Reply:
edit: I meant create(), insert() doesn't exist as a valid prisma method. 
If you want to update if there is already an item present, you need to fill this update object